src/lib.rs

```diff
@@ -89,11 +89,53 @@ impl<K: std::hash::Hash + PartialEq, V> HashTable<K, V> {
     }
 }
 
+pub struct HashTableIterator<'a, K, V> {
+    elements_iterator: Box<dyn Iterator<Item = &'a (K, V)> + 'a>,
+    buckets_iterator: Box<dyn Iterator<Item = &'a Vec<(K, V)>> + 'a>,
+}
+
+impl<'a, K, V> IntoIterator for &'a HashTable<K, V> {
+    type Item = &'a (K, V);
+
+    type IntoIter = HashTableIterator<'a, K, V>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let mut buckets_iterator = self.buckets.iter();
+        // first elements iterator needs to be initialized
+        let elements_iterator = buckets_iterator
+            .next()
+            .map(|bi| bi.iter())
+            .unwrap_or_else(|| [].iter());
```

Comment:
Still odd to me. Related: https://stackoverflow.com/questions/32284362/how-to-create-an-empty-iterator-for-a-certain-collection-type-list-set-map-in